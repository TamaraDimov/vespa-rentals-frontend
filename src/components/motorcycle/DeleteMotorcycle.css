/* Delete motorcycle page styles

/* Page container */
.delete-motorcycle-bg {
  background-color: #e35e0a;
  background-image: url('../../assets/Delete_bg.jpg');
  background-size: cover;
  background-position: center;
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
}

/* Container for motorcycle list */
.delete-motorcycle-container {
  background-color: rgba(255, 255, 255, 0.9);
  padding: 1rem;
  border-radius: 10px;
  box-shadow: 0 0 10px rgba(0, 0, 0, 0.2);
  max-width: 650px;
  width: 100%;
}

/* Heading */
h1 {
  font-size: 2.5rem;
  font-weight: bold;
  margin-bottom: 30px;
  text-align: center;
}

/* Message */
p {
  font-size: 1.2rem;
  margin-bottom: 30px;
  text-align: center;
}

/* Motorcycle list */
.delete-motorcycle-list-all {
  display: flex;
  justify-content: space-between;
  align-items: center;
  list-style: none;
  text-align: center;
}

/* Motorcycle item */

.delete-motorcycle-list {
  display: flex;
  align-self: center;
  justify-content: center;
  width: 100%;
  padding: 3% 2%;
}

.delete-motorcycle-list li img {
  width: 100%;
  height: 30vh;
  border-radius: 10px;
  border: 4px solid #e35e0a;
  box-shadow: 0 0 5px #ddb49b, 0 0 10px #ec8849, 0 0 20px #e35e0a, 0 0 40px;
  animation: glow 1s ease-in-out infinite alternate;
}

@keyframes glow {
  from {
    box-shadow:
      0 0 2px #d89b74,
      0 0 8px #f47e35,
      0 0 10px #e35e0a,
      0 0 20px #e35e0a;
  }

  to {
    box-shadow:
      0 0 5px #e35e0a,
      0 0 10px #e35e0a,
      0 0 15px #e35e0a,
      0 0 20px #e35e0a;
  }
}

.delete-motorcycle-list li h2 {
  font-size: 1.5rem;
  font-weight: bold;
  margin-top: 20px;
  margin-bottom: 10px;
}

.delete-motorcycle-list li p {
  font-size: 1.2rem;
  margin-bottom: 20px;
}

.del-btn-danger {
  background-color: #fff;
  color: #e35e0a;
  border: none;
  border-radius: 5px;
  padding: 10px 20px;
  font-size: 1.2rem;
  font-weight: bold;
  cursor: pointer;
  transition: background-color 0.2s ease-in-out;
  bottom: 20px;
  left: 50%;
}

.del-btn-danger:hover {
  background-color: #e3200a;
  color: #fff;
}

@media (max-width: 768px) {
  .delete-motorcycle-container {
    padding: 50px;
  }

  .delete-motorcycle-container h1 {
    font-size: calc(0.8rem + 2vw);
    font-weight: bold;
    margin-bottom: 1vw;
    text-align: center;
  }
}
