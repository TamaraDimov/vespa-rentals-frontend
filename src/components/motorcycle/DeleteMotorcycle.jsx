import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { ChevronLeft, ChevronRight } from 'react-feather';
import { fetchMotorcycle } from '../../redux/reducers/motorcycleSlice';
import PopDelete from './subcomponents/pop/Delete';
import './DeleteMotorcycle.css';

const DeleteMotorcycle = () => {
  const motorcycles = useSelector((state) => state.motorcycle);
  const { user } = useSelector((state) => state.user);
  const [confirm, setConfirm] = useState(false);
  const [id, setId] = useState();
  const { token } = user.user;
  const [curr, setCurr] = useState(0);
  const dispatch = useDispatch();
  const prev = () => setCurr((curr) => (curr === 0 ? motorcycles.motorcycle.length - 1 : curr - 1));
  const next = () => setCurr((curr) => (curr === motorcycles.motorcycle.length - 1 ? 0 : curr + 1));
  useEffect(() => {
    dispatch(fetchMotorcycle(token));
  }, [dispatch, token]);
  const handleConfirmation = () => {
    setConfirm(!confirm);
  };
  const available = motorcycles.motorcycle.length;
  const message = `We have ${available} motorcycle${
    available !== 1 ? 's' : ''
  } available.
  ${
  available
    ? 'Select a motorcycle down below to delete!'
    : 'No motorcycles are available!'
}`;
  return (
    <section className="delete-motorcycle-bg">
      <div className="delete-motorcycle-container overflow-hidden relative">
        {confirm && (
          <PopDelete
            id={id}
            confirm={confirm}
            setConfirm={setConfirm}
            setCurr={setCurr}
          />
        )}
        <h1>Available motorcycles</h1>
        <p className="text-slate-800 hover:text-black-600">{message}</p>
        <div
          className="delete-motorcycle-list-all flex transition-transform ease-out duration-500"
          style={{ transform: `translateX(-${curr * 100}%)` }}
        >
          {Array.isArray(motorcycles.motorcycle)
            && motorcycles.motorcycle.map((motorcycle, index) => (
              <div
                key={motorcycle.id}
                className={`delete-motorcycle-list flex-none w-full ${
                  index !== curr
                    ? 'hidden opacity-0 transition-opacity duration-500'
                    : 'opacity-100 transition-opacity duration-500'
                }`}
              >
                <li>
                  <img
                    src={motorcycle.photo}
                    alt={motorcycle.model}
                    className="w-full h-full object-cover"
                  />
                  <div>
                    <h2>{motorcycle.name}</h2>
                    <p>{motorcycle.description}</p>
                    <button
                      type="button"
                      className="del-btn-danger z-10"
                      onClick={() => {
                        setId(motorcycle.id);
                        handleConfirmation();
                        window.scrollTo(0, 0);
                      }}
                    >
                      Delete
                    </button>
                  </div>
                </li>
              </div>
            ))}
        </div>
        <div className="inset-0 flex items-center justify-between p-3">
          <button
            type="button"
            onClick={prev}
            className="p-1 rounded-full shadow bg-black/80 text-white hover:bg-gray-500"
          >
            <ChevronLeft />
          </button>
          <button
            type="button"
            onClick={next}
            className="p-1 rounded-full shadow bg-black/80 text-white hover:bg-gray-500"
          >
            <ChevronRight />
          </button>
        </div>
        <div className="absolute bottom-4 right-0 left-0">
          <div className="flex items-center justify-center gap-2">
            {Array.isArray(motorcycles.motorcycle)
              && motorcycles.motorcycle.map((motorcycle, i) => (
                <div
                  key={motorcycle.id}
                  className={`
              transition-all w-3 h-3 bg-black rounded-full
              ${curr === i ? 'p-2' : 'bg-opacity-50'}
            `}
                />
              ))}
          </div>
        </div>
      </div>
    </section>
  );
};
export default DeleteMotorcycle;
