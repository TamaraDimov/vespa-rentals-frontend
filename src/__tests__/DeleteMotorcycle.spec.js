import '@testing-library/jest-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import DeleteMotorcycle from '../components/motorcycle/DeleteMotorcycle';
import 'mock-local-storage';

const mockStore = configureMockStore();
const testuser = {
  user: {
    data: {
      id: 1,
      email: 'kiflekifle23@gmail.com',
      created_at: '2023-09-22T13:51:50.089Z',
    },
    token:
      'eyJhbGciOiJIUzI1NiJ9.eyJqdGkiOiJlYjI2ZTA2Yy1lMTM1LTRmMmEtYTYwZi1iYmMwM2I1MGE2MmQiLCJzdWIiOiIxIiwic2NwIjoidXNlciIsImF1ZCI6bnVsbCwiaWF0IjoxNjk2MjI2NTUwLCJleHAiOjE3MDE0MTA1NTB9.g0XLlQ7sc-1r0KIevNXVKwwH61FzdXM_sB-tpS6WWK4',
  },
};
const store = mockStore({
  motorcycle: {
    motorcycle: [
      {
        id: 1,
        name: 'Honda CBR600RR',
        model: 'CBR600RR',
        photo: 'https://example.com/photo1.jpg',
        description: 'Description for Honda CBR600RR',
      },
      {
        id: 2,
        name: 'Yamaha YZF-R6',
        model: 'YZF-R6',
        photo: 'https://example.com/photo2.jpg',
        description: 'Description for Yamaha YZF-R6',
      },
    ],
  },
  user: {
    isLoading: false,
    user: testuser,
    token: testuser.token,
  },
});
describe('DeleteMotorcycle', () => {
  let mockLocalStorage;
  let originalLocalStorage;

  beforeAll(() => {
    originalLocalStorage = global.localStorage;

    mockLocalStorage = {
      getItem: jest.fn(),
      setItem: jest.fn(),
      removeItem: jest.fn(),
      clear: jest.fn(),
    };

    const localStorageMock = {
      ...mockLocalStorage,
      getItem: jest.fn().mockImplementation((key) => mockLocalStorage[key]),
      setItem: jest.fn().mockImplementation((key, value) => {
        mockLocalStorage[key] = value;
      }),
      removeItem: jest.fn().mockImplementation((key) => {
        delete mockLocalStorage[key];
      }),
      clear: jest.fn().mockImplementation(() => {
        mockLocalStorage = {};
      }),
    };

    Object.defineProperty(global, 'localStorage', {
      value: localStorageMock,
    });

    localStorageMock.getItem.mockImplementation(() => JSON.stringify(testuser));
  });

  afterAll(() => {
    Object.defineProperty(global, 'localStorage', {
      value: originalLocalStorage,
    });
  });

  let deleteMotorcycle;
  beforeAll(async () => {
    deleteMotorcycle = renderer
      .create(
        <Provider store={store}>
          <Router>
            <DeleteMotorcycle />
          </Router>
        </Provider>,
      )
      .toJSON();
  });

  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('should render correctly', () => {
    expect(deleteMotorcycle).toBeTruthy();
  });

  it('should match the snapshot', () => {
    expect(deleteMotorcycle).toMatchSnapshot();
  });
});
