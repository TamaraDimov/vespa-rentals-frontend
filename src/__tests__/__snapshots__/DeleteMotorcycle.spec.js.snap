// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteMotorcycle should match the snapshot 1`] = `
<section
  className="delete-motorcycle-bg"
>
  <div
    className="delete-motorcycle-container overflow-hidden relative"
  >
    <h1>
      Available motorcycles
    </h1>
    <p
      className="text-slate-800 hover:text-black-600"
    >
      We have 2 motorcycles available.
  Select a motorcycle down below to delete!
    </p>
    <div
      className="delete-motorcycle-list-all flex transition-transform ease-out duration-500"
      style={
        Object {
          "transform": "translateX(-0%)",
        }
      }
    >
      <div
        className="delete-motorcycle-list flex-none w-full opacity-100 transition-opacity duration-500"
      >
        <li>
          <img
            alt="CBR600RR"
            className="w-full h-full object-cover"
            src="https://example.com/photo1.jpg"
          />
          <div>
            <h2>
              Honda CBR600RR
            </h2>
            <p>
              Description for Honda CBR600RR
            </p>
            <button
              className="del-btn-danger z-10"
              onClick={[Function]}
              type="button"
            >
              Delete
            </button>
          </div>
        </li>
      </div>
      <div
        className="delete-motorcycle-list flex-none w-full hidden opacity-0 transition-opacity duration-500"
      >
        <li>
          <img
            alt="YZF-R6"
            className="w-full h-full object-cover"
            src="https://example.com/photo2.jpg"
          />
          <div>
            <h2>
              Yamaha YZF-R6
            </h2>
            <p>
              Description for Yamaha YZF-R6
            </p>
            <button
              className="del-btn-danger z-10"
              onClick={[Function]}
              type="button"
            >
              Delete
            </button>
          </div>
        </li>
      </div>
    </div>
    <div
      className="inset-0 flex items-center justify-between p-3"
    >
      <button
        className="p-1 rounded-full shadow bg-black/80 text-white hover:bg-gray-500"
        onClick={[Function]}
        type="button"
      >
        <svg
          fill="none"
          height={24}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="15 18 9 12 15 6"
          />
        </svg>
      </button>
      <button
        className="p-1 rounded-full shadow bg-black/80 text-white hover:bg-gray-500"
        onClick={[Function]}
        type="button"
      >
        <svg
          fill="none"
          height={24}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="9 18 15 12 9 6"
          />
        </svg>
      </button>
    </div>
    <div
      className="absolute bottom-4 right-0 left-0"
    >
      <div
        className="flex items-center justify-center gap-2"
      >
        <div
          className="
              transition-all w-3 h-3 bg-black rounded-full
              p-2
            "
        />
        <div
          className="
              transition-all w-3 h-3 bg-black rounded-full
              bg-opacity-50
            "
        />
      </div>
    </div>
  </div>
</section>
`;
